### PR TITLE
Support bastion HA with external vip

### PR DESCRIPTION
**Pull request by yussufsh**
Signed-off-by: Yussuf Shaikh <yussuf.shaikh@ibm.com>

### CODE REVIEW

**Comment by ltccci** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/ocp-power-automation/ocp4-upi-powervs/pull/114#" title="Author self-approved">yussufsh</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=ocp-power-automation%2Focp4-upi-powervs).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/ocp-power-automation/ocp4-upi-powervs/blob/master/OWNERS)~~ [yussufsh]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by yussufsh** [comment]
/cc @Prajyot-Parab @bpradipt 
I know this is a big change, but I took the time to divide it into logical commits for your review.

File: modules/5_install/install.tf
Comment (bpradipt) [inline modules/5_install/install.tf:22]: Better to add a comment like "FIXME" when api is available

File: modules/5_install/templates/install_inventory
Comment (bpradipt) [inline modules/5_install/templates/install_inventory:3]: Should the `ansible_user` come from the `rhel_username` variable used in var.tfvars ?

File: variables.tf
Comment (bpradipt) [inline variables.tf:57]: I'm wondering do we really need 3? Since it's just active/passive we can restrict the count to 2. Thoughts ?

File: modules/5_install/templates/install_inventory
Comment (yussufsh) [reply modules/5_install/templates/install_inventory:3]: Yes.. will create an issue for this to be fixed across repos.

Comment (yussufsh) [reply modules/5_install/templates/install_inventory:3]: https://github.com/ocp-power-automation/ocp4-upi-powervs/issues/116

Comment (yussufsh) [reply modules/5_install/templates/install_inventory:3]: Will take it in new PR. Okay?

File: variables.tf
Comment (yussufsh) [reply variables.tf:57]: Even I was wondering what should be the ideal value. After evaluating the technical limitation on no of external IP addresses in pub-vlan I have concluded 3 can be done,

But I guess yes maximum `2` make sense with less cost and management. I will change it to 2 but if anyone asks for a new CR we can just update the number here.. the code is made to handle N no of bastion nodes.

File: modules/5_install/templates/install_inventory
Comment (bpradipt) [reply modules/5_install/templates/install_inventory:3]: yeah that should be fine.

File: variables.tf
Comment (bpradipt) [reply variables.tf:57]: makes sense. Let's have 2 as max and if we really get a usecase for more than that it can be changed

**Comment by ltccci** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/ocp-power-automation/ocp4-upi-powervs/pull/114#issuecomment-706108831" title="Approved">bpradipt</a>*, *<a href="https://github.com/ocp-power-automation/ocp4-upi-powervs/pull/114#" title="Author self-approved">yussufsh</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=ocp-power-automation%2Focp4-upi-powervs).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/ocp-power-automation/ocp4-upi-powervs/blob/master/OWNERS)~~ [bpradipt,yussufsh]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Review by bpradipt — DISMISSED** [review]

**Comment by ltccci** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/ocp-power-automation/ocp4-upi-powervs/pull/114#pullrequestreview-505561541" title="Approved">bpradipt</a>*, *<a href="https://github.com/ocp-power-automation/ocp4-upi-powervs/pull/114#" title="Author self-approved">yussufsh</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=ocp-power-automation%2Focp4-upi-powervs).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/ocp-power-automation/ocp4-upi-powervs/blob/master/OWNERS)~~ [bpradipt,yussufsh]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: modules/5_install/templates/keepalived_vrrp_instance.tpl
Comment (Prajyot-Parab) [inline modules/5_install/templates/keepalived_vrrp_instance.tpl:5]: @yussufsh Seems the same priority will be set on all the nodes, as per keepalived guidelines we should have lower priority on Backup nodes compared to Master.

Comment (yussufsh) [reply modules/5_install/templates/keepalived_vrrp_instance.tpl:5]: Good point @Prajyot-Parab  I will push the changes required for public vrrp.

**Comment by ltccci** [comment]
New changes are detected. LGTM label has been removed.

**Comment by ltccci** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/ocp-power-automation/ocp4-upi-powervs/pull/114#issuecomment-706108831" title="Approved">bpradipt</a>*, *<a href="https://github.com/ocp-power-automation/ocp4-upi-powervs/pull/114#" title="Author self-approved">yussufsh</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=ocp-power-automation%2Focp4-upi-powervs).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/ocp-power-automation/ocp4-upi-powervs/blob/master/OWNERS)~~ [bpradipt,yussufsh]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by ltccci** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/ocp-power-automation/ocp4-upi-powervs/pull/114#issuecomment-706108831" title="Approved">bpradipt</a>*, *<a href="https://github.com/ocp-power-automation/ocp4-upi-powervs/pull/114#" title="Author self-approved">yussufsh</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=ocp-power-automation%2Focp4-upi-powervs).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/ocp-power-automation/ocp4-upi-powervs/blob/master/OWNERS)~~ [bpradipt,yussufsh]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Review by Prajyot-Parab — APPROVED** [review]
/lgtm

**Comment by ltccci** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/ocp-power-automation/ocp4-upi-powervs/pull/114#issuecomment-706108831" title="Approved">bpradipt</a>*, *<a href="https://github.com/ocp-power-automation/ocp4-upi-powervs/pull/114#pullrequestreview-505842505" title="LGTM">Prajyot-Parab</a>*, *<a href="https://github.com/ocp-power-automation/ocp4-upi-powervs/pull/114#" title="Author self-approved">yussufsh</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=ocp-power-automation%2Focp4-upi-powervs).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/ocp-power-automation/ocp4-upi-powervs/blob/master/OWNERS)~~ [Prajyot-Parab,bpradipt,yussufsh]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Review by bpradipt — APPROVED** [review]

**Comment by ltccci** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/ocp-power-automation/ocp4-upi-powervs/pull/114#pullrequestreview-505867712" title="Approved">bpradipt</a>*, *<a href="https://github.com/ocp-power-automation/ocp4-upi-powervs/pull/114#pullrequestreview-505842505" title="LGTM">Prajyot-Parab</a>*, *<a href="https://github.com/ocp-power-automation/ocp4-upi-powervs/pull/114#" title="Author self-approved">yussufsh</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=ocp-power-automation%2Focp4-upi-powervs).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/ocp-power-automation/ocp4-upi-powervs/blob/master/OWNERS)~~ [Prajyot-Parab,bpradipt,yussufsh]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by ssinha20** [comment]
@yussufsh : Since the PR is merged, can the issue be closed or do you plan further development?

**Comment by yussufsh** [comment]
> @yussufsh : Since the PR is merged, can the issue be closed or do you plan further development?

Wrong PR?

**Comment by ssinha20** [comment]
https://github.com/ocp-power-automation/ocp4-upi-powervs/issues/116 - is this the correct issue linked with this PR?

**Comment by yussufsh** [comment]
> #116 - is this the correct issue linked with this PR?

No

**Pull request merged**
